Common/src/geometry_structure.cpp

```diff
@@ -14957,6 +14958,34 @@ void CPhysicalGeometry::SetBoundControlVolume(CConfig *config, unsigned short ac
   
 }
 
+void CPhysicalGeometry::SetMaxLength(void) {
```

Comment:
The computation in this routine has two major issues. First of all, if run in parallel it is not taking into account that there are neighbors on other processes. So a communication is necessary. Second, if using AD, the computation of a maximum generates a lot of unnecessary overhead since we store all computations of delta_aux, although we only need the maximum value. I know it is not very intuitive, but one way to solve that is to do the computation of the maximum first using the passivedouble datatype to find out the iNeigh index. Then, redo the computation of the delta_aux using su2double datatype.

Let me know if you need help.

Reply:
@talbring Thanks for pointing out my mistakes. This PR is even more important then, because those problems are also present in the original code (i.e., the `develop` branch).  The max calculations are just moved to a new place in this PR--I didn't change the underlying algorithm.

I should be able to fix the MPI communication myself.

For the AD portion, I can commit some changes and then have you review them.  I haven't worked much with the AD side of SU2, so my knowledge is limited.

Reply:
@talbring Can you take a look and see if my recent commits have fixed the two problems you identified?